### DISCUSSION TITLE
What are the params to be passed for Salesforce Connection Type

### DISCUSSION THREAD

**Discussion by kgowrisangar** [schemacrawler/SchemaCrawler#497] [Q&A] [answered]
With the help of @sualeh , identified that there's a way to use the salesforce jdbc [driver ](https://www.cdata.com/drivers/salesforce/download/jdbc/) to crawl salesforce data. But I'd need to know what would be the params that should be used while calling the schemacrawler.sh ? 

For eg from something like this, what're the params I should use:

`schemacrawler.sh --server="$1" --host="$2" --database="$3" --user="$4" --password="$5" --port="$6" --log-level=INFO --info-level=standard -c script --sort-tables=false --script "$7"`

Any help would be appreciated.

**Comment by sualeh** [comment, marked as answer]
You will need to use the [CData Salesforce JDBC driver](https://www.cdata.com/drivers/salesforce/jdbc/). You will need to put the driver jar file (along with all associated dependencies) in the SchemaCrawler "lib" folder.

You will need to use the URL version of the command-line:
```sh
schemacrawler.sh \
--url="$1" \
--user="$2" \
--password="$3" \
--log-level=INFO \
--info-level=standard \
-c script \
--sort-tables=false \
--script "$4"
```
Please read the [documentation ](https://cdn.cdata.com/help/RFG/jdbc/pg_JDBCconnectcode.htm)on how to get the URL. 

I would suggest testing it first with the "list" command:
```sh
schemacrawler.sh \
--url="$1" \
--user="$2" \
--password="$3" \
--log-level=INFO \
--info-level=minimum \
-c list
```

**Reply by kgowrisangar** [reply]
Thank you @sualeh  Will try this out and keep you posted

**Comment by sualeh** [comment]
@kgowrisangar - any updates? Did that work for you?

**Reply by kgowrisangar** [reply]
Apologies for the delay @sualeh. Thanks it worked

**Reply by sualeh** [reply]
@kgowrisangar - time to publish another dev.to article with your experiences. 😄
